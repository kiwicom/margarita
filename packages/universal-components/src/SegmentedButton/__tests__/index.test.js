// @flow

import * as React from 'react';
import { render, shallow, fireEvent } from 'react-native-testing-library';

import { SegmentedButton } from '..';

describe('SegmentedButton', () => {
  const testData = [{ label: 'A', value: 'a' }, { label: 'B', value: 'b' }];
  const onValueChange = jest.fn();

  const { getByText } = render(
    <SegmentedButton
      segmentsData={testData}
      selectedValue={testData[0].value}
      onValueChange={onValueChange}
    />,
  );

  it('should contain labels', () => {
    expect(getByText(testData[0].label)).toBeDefined();
    expect(getByText(testData[1].label)).toBeDefined();
  });

  it('should execute onValueChange method with correct value', () => {
    fireEvent(getByText(testData[0].label), 'press');
    expect(onValueChange).toHaveBeenCalledWith(testData[0].value);
  });

  it('renders', () => {
    expect(
      shallow(
        <SegmentedButton
          segmentsData={testData}
          selectedValue="a"
          onValueChange={jest.fn()}
        />,
      ),
    ).toMatchSnapshot();
  });
});
