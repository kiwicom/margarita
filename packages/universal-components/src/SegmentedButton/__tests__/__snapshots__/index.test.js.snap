// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedButton renders 1`] = `
Object {
  "output": <Component
    style={
      Object {
        "backgroundColor": "#e8edf1",
        "borderRadius": 3,
        "flexDirection": "row",
        "padding": 2,
      }
    }
  >
    <Segment
      active={true}
      first={true}
      label="A"
      last={false}
      onPress={[MockFunction]}
      value="a"
    />
    <Segment
      active={false}
      first={false}
      label="B"
      last={true}
      onPress={[MockFunction]}
      value="b"
    />
  </Component>,
}
`;
